R/recipe.R

```diff
@@ -593,24 +622,28 @@ bake <- function(object, ...) {
 #' @export
 bake.recipe <- function(object, new_data, ..., composition = "tibble") {
   if (rlang::is_missing(new_data)) {
-    rlang::abort("'new_data' must be either a data frame or NULL. No value is not allowed.")
+    cli::cli_abort(
+      "{.arg new_data} must be either a data frame or NULL. \\
+      No value is not allowed."
+    )
   }
 
   if (is.null(new_data)) {
     return(juice(object, ..., composition = composition))
   }
 
   if (!fully_trained(object)) {
-    rlang::abort("At least one step has not been trained. Please run `prep`.")
+    cli::cli_abort(c(
+      "x" = "At least one step has not been trained.",
+      "i" = "Please run {.fun recipes::prep}."
+    ))
   }
 
   if (!any(composition == formats)) {
-    rlang::abort(
-      paste0(
-        "`composition` should be one of: ",
-        paste0("'", formats, "'", collapse = ",")
-      )
-    )
+    cli::cli_abort(c(
+      "x" = "{.arg composition} cannot be {.val composition}.",
+      "i" = "Allowed values are {.or formats}."
```

Comment:
I believe we need to bracket-substitute in a couple places here:

``` r
composition <- "hey"
formats <- "there"

cli::cli_abort(c(
  "x" = "{.arg composition} cannot be {.val composition}.",
  "i" = "Allowed values are {.or formats}."
))
#> Error:
#> ✖ `composition` cannot be "composition".
#> ℹ Allowed values are formats.
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(...)
#>  2.   └─rlang::abort(...)
```

<sup>Created on 2023-11-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
you are correct!

Reply:
done in [use variable substitution](https://github.com/tidymodels/recipes/pull/1251/commits/d57ae17ce0683cb6fabfbc6adbff579d54538421)